diff --git a/wrappers/python/CMakeLists.txt b/wrappers/python/CMakeLists.txt
index 2abc3bf..d8ae896 100644
--- a/wrappers/python/CMakeLists.txt
+++ b/wrappers/python/CMakeLists.txt
@@ -10,7 +10,7 @@ list(SORT Header_Files)
 list(SORT Source_Files)
 list(SORT Python_Files)
 
-pybind11_add_module(pyodil SHARED ${Source_Files} ${Header_Files})
+pybind11_add_module(pyodil ${Source_Files} ${Header_Files})
 
 target_compile_definitions(
     pyodil PRIVATE $<$<BOOL:BUILD_SHARED_LIBS>:BOOST_ALL_DYN_LINK>)
