From f8b2c81affde82791a67dc6058f8d24e3b697809 Mon Sep 17 00:00:00 2001
From: Julien Lamy <lamy@unistra.fr>
Date: Mon, 5 Feb 2024 11:40:33 +0100
Subject: [PATCH 1/4] Fix module type

---
 wrappers/python/CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/wrappers/python/CMakeLists.txt b/wrappers/python/CMakeLists.txt
index 2abc3bf..1d56c8c 100644
--- a/wrappers/python/CMakeLists.txt
+++ b/wrappers/python/CMakeLists.txt
@@ -10,7 +10,7 @@ list(SORT Header_Files)
 list(SORT Source_Files)
 list(SORT Python_Files)
 
-pybind11_add_module(pyodil SHARED ${Source_Files} ${Header_Files})
+pybind11_add_module(pyodil ${Source_Files} ${Header_Files})
 
 target_compile_definitions(
     pyodil PRIVATE $<$<BOOL:BUILD_SHARED_LIBS>:BOOST_ALL_DYN_LINK>)
@@ -38,3 +38,7 @@ execute_process(
 
 install(TARGETS pyodil DESTINATION "${PYTHON_SITE_PACKAGES}/odil")
 install(FILES ${Python_Files} DESTINATION "${PYTHON_SITE_PACKAGES}/odil")
+
+if(WIN32)
+    install(TARGETS libodil RUNTIME DESTINATION "${PYTHON_SITE_PACKAGES}/odil")
+endif()
